agent.py

```
@@ -0,0 +1,1114 @@
+            self.target_server = os.getenv("WINDOWS_TARGET_SERVER", "localhost")
+        self.hostname = socket.gethostname() if self.target_server == "localhost" else self.target_server
+    
+    def _execute_powershell_command(self, command: str, remote: bool = False) -> str:
+        try:
+            if remote and self.target_server != "localhost":
+                cmd = (
+                    f"powershell -Command \"Invoke-Command -ComputerName {self.target_server} -ScriptBlock {{{command}}}\""
+                )
+            else:
+                cmd = f"powershell -Command \"{command}\""
+            result = subprocess.run(cmd, shell=True, capture_output=True, text=True)
```

Ruff (0.13.1)

`subprocess` call with `shell=True` identified, security issue

(S602)